kmath-core/src/commonMain/kotlin/space/kscience/kmath/operations/algebraExtensions.kt

```diff
@@ -135,7 +135,7 @@ private fun <T> Ring<T>.powWithoutOptimization(base: T, exponent: UInt): T = whe
  * @return the base raised to the power.
  * @author Iaroslav Postovalov, Evgeniy Zhelenskiy
  */
-public fun <T> Field<T>.power(arg: T, power: Int): T = when {
-    power < 0 -> one / (this as Ring<T>).power(arg, if (power == Int.MIN_VALUE) Int.MAX_VALUE.toUInt().inc() else (-power).toUInt())
-    else -> (this as Ring<T>).power(arg, power.toUInt())
+public fun <T> Field<T>.power(arg: T, power: UInt): T = when {
```

Comment:
@altavir You are wrong here: Field can have negative power as Field has division. That was even in the previous implementation.

Reply:
It is reverted.